master/cmd/determined-master/root.go

```diff
@@ -137,6 +143,47 @@ func getConfig(configMap map[string]interface{}) (*internal.Config, error) {
 	return config, nil
 }
 
+func applyBindMount(configMap map[string]interface{}) (map[string]interface{}, error) {
+	bindMountField, fieldExists := configMap["auto_bind_mount"]
+	if !fieldExists {
+		return configMap, nil
+	}
+	vBindMount, ok := bindMountField.(string)
+	if !ok {
+		return nil, errors.New("wrong type for bind mount field")
+	}
+
+	if vBindMount != "" {
+		tcd, tcdExisted := configMap["task_container_defaults"]
+		if !tcdExisted {
+			newTCD := make(map[string]interface{})
+			bindMounts := []model.BindMount{
+				{
+					HostPath:      vBindMount,
+					ContainerPath: "/run/determined/shared_fs",
+				},
+			}
+			newTCD["bind_mounts"] = bindMounts
+			configMap["task_container_defaults"] = newTCD
+		} else {
+			vTCD, ok := tcd.(map[string]interface{})
+			if !ok {
+				return nil, errors.New("wrong type for task_container_defaults field")
+			}
+
+			vTCD["bind_mounts"] = append(
+				vTCD["bind_mounts"].([]model.BindMount),
+				model.BindMount{
+					HostPath:      vBindMount,
+					ContainerPath: "/run/determined/shared_fs",
```

Comment:
@naren-determined is this the long-term location we expect to mount this directory to?

Reply:
@shiyuann can confirm, but this is also the mounting location for AWS/GCP for notebooks.

Reply:
@rb-determined-ai yes. Before we want to change, this will be the path for the shared fs container path